### PR TITLE
PHOENIX-7343: Support ARRAY and JSON types as simple values in CDC

### PR DESCRIPTION
**Pull request by haridsv**
Jira: PHOENIX-7343

Added support for complex types ARRAY and JSON so that they are treated at simple values. As they are treated as simple values, changes with n the value will not be detected.

### CODE REVIEW

**Review by ranganathg — APPROVED** [review]

File: phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java [hunk 1]
```diff
@@ -257,7 +252,7 @@ private Object getColumnValue(Cell cell, PDataType dataType) {
 
     private Object getColumnValue(byte[] cellValue, int offset, int length, PDataType dataType) {
         Object value;
-        if (dataType.getSqlType() == Types.BINARY) {
+        if (CDCUtil.isBinaryType(dataType)) {
```
Comment (virajjasani) [inline phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java:255]: This is an improvement with this PR, correct?

**Review by virajjasani — APPROVED** [review]
Left one question, +1 otherwise

File: phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java (hunk 1 above)
Comment (haridsv) [reply phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java:255]: This is just a utility method to abstract out the 3 types that we are considering as binary type.

Comment (virajjasani) [reply phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java:255]: Oh i missed that it was Types.BINARY which anyways apply to them, i think i misunderstood earlier that we only had Binary datatype, and not VARBINARY.

**Comment by virajjasani** [comment]
@haridsv i am about to merge the PR but just realized the PR has merge conflict, that most likely came from today's changes.

**Comment by haridsv** [comment]
Yes, looks like there is a new binary type added to the condition, I will resolve the conflict.

**Review by virajjasani — APPROVED** [review]
+1, pending jenkins build

**Pull request merged**